scrapy/http/response/json.py

```diff
@@ -0,0 +1,16 @@
+"""
+This module implements the JsonResponse class which adds encoding
+discovering through JSON encoding declarations to the TextResponse class.
+
+See documentation in docs/topics/request-response.rst
+"""
+
+import json
+from scrapy.http.response.text import TextResponse
+
+
+class JsonResponse(TextResponse):
+
+    def json(self):
```

Comment:
Should we be storing the results of the `json.loads` call, to avoid performing the same operation multiple times? Something like
```python
try:
    return self._json
except AttributeError:
    self._json = json.loads(self.text)
    return self._json
```
or maybe
```python
if not hasattr(self, "_json"):
    self._json = json.loads(self.text)
return self._json
```

Reply:
Or we can use the following check like the `text()` method in `TextResponse` :
```
if self._json is None:
    self._json = json.loads(self.text)
return self._json
```

Reply:
Nicely done, I wasn't aware of that. Can I request a last small adjustment, to keep things consistent? Could you rename the attribute to something like `_cached_decoded_json`?